front/src/applications/operationalStudies/components/MacroEditor/OsrdNgeSync.tsx

```diff
@@ -0,0 +1,918 @@
+            sourceDeparture = createTimeLock(
+              addDurationToDate(
+                addDurationToDate(startTime, sourceScheduleEntry.arrival),
+                sourceScheduleEntry.stop_for || 'P0D'
+              )
+            );
+          }
+
+          let targetArrival = { ...DEFAULT_TIME_LOCK };
+          if (targetScheduleEntry && targetScheduleEntry.arrival) {
+            targetArrival = createTimeLock(
+              addDurationToDate(startTime, targetScheduleEntry.arrival)
+            );
+          }
+
+          const travelTime = { ...DEFAULT_TIME_LOCK };
+          if (targetArrival.consecutiveTime !== null && sourceDeparture.consecutiveTime !== null) {
+            travelTime.time = targetArrival.consecutiveTime - sourceDeparture.consecutiveTime;
+            travelTime.consecutiveTime = travelTime.time;
+          }
+
+          const trainrunSection = {
+            id: trainrunSectionId,
+            sourceNodeId: sourceNode.id,
+            sourcePortId: sourcePort.id,
+            targetNodeId: targetNode.id,
+            targetPortId: targetPort.id,
+            travelTime,
+            sourceDeparture,
+            sourceArrival: { ...DEFAULT_TIME_LOCK },
+            targetDeparture: { ...DEFAULT_TIME_LOCK },
+            targetArrival,
+            numberOfStops: 0,
+            trainrunId: trainSchedule.id,
+            resourceId: this.ngeResource.id,
+            path: {
+              path: [],
+              textPositions: [],
+            },
+            specificTrainrunSectionFrequencyId: 0,
+            warnings: [],
+          };
+
+          trainrunSectionId += 1;
+          return trainrunSection;
+        });
+      })
+      .flat();
+
+    return {
+      trainrunSections,
+      nodes: Object.values(ngeNodesByPathKey),
+    };
+  }
+
+  /**
+   * Execute the search payload and collect all result pages.
+   */
+  private async executeSearch(): Promise<SearchResultItemOperationalPoint[]> {
+    const searchResults: SearchResultItemOperationalPoint[] = [];
+    const searchPayload = this.buildOpQuery();
+    if (searchPayload) {
+      const pageSize = 100;
+      let done = false;
+      for (let page = 1; !done; page += 1) {
+        const searchPromise = this.dispatch(
+          osrdEditoastApi.endpoints.postSearch.initiate(
+            {
+              page,
+              pageSize,
+              searchPayload,
+            },
+            { track: false }
+          )
+        );
+        const results = (await searchPromise.unwrap()) as SearchResultItemOperationalPoint[];
+        searchResults.push(...results);
+        done = results.length < pageSize;
+      }
+    }
+    return searchResults;
+  }
+
+  /**
+   * Build a search query to fetch all operational points from their UICs,
+   * trigrams and IDs.
+   */
+  private buildOpQuery(): SearchPayload | null {
+    const pathItems = this.trainSchedules.map((schedule) => schedule.path).flat();
```

Comment:
- rename schedule by train
- why not flatMap ?
```suggestion
    const pathItems = this.trainSchedules.flatMap((train) => train.path);
```